client/apps/edit/components/content/sections/image_collection/components/controls.jsx

```diff
@@ -1,186 +1,197 @@
-            vals.push({
-              id,
-              value: r.title,
-              thumbnail: (r._links.thumbnail != null ? r._links.thumbnail.href : null)
-            })
-          }
-        }
-        return vals
-      },
-      templates: {
-        suggestion (data) {
-          return `<div class='autocomplete-suggestion' style='background-image: url(${data.thumbnail})'></div>${data.value}`
+  filterAutocomplete = (items) => {
+    return items._embedded.results.map((item) => {
+      const { type } = item
+
+      if (type === 'artwork') {
+        const { title, _links } = item
+        const { thumbnail, self } = _links
+        const _id = self.href.substr(self.href.lastIndexOf('/') + 1)
+        const thumbnail_image = thumbnail && thumbnail.href
+
+        return {
+          _id,
+          title,
+          thumbnail_image,
+          type
         }
-      },
-      selected: this.onSelectArtwork
+      } else {
+        return false
+      }
     })
-    return defer(() => $el.focus())
   }
 
-  onSelectArtwork = (e, selected) => {
-    new Artwork({id: selected.id}).fetch({
-      success: (artwork) => {
-        let newImages = clone(this.props.images)
-        newImages = newImages.concat([artwork.denormalized()])
-        this.props.section.set('images', newImages)
-        $(this.refs.autocomplete).val('').focus()
-        this.props.onChange()
-      }
-    })
+  fetchDenormalizedArtwork = async (id) => {
+    const { logErrorAction } = this.props
+
+    try {
+      const artwork = await new Artwork({ id }).fetch()
+      return new Artwork(artwork).denormalized()
+    } catch (err) {
+      logErrorAction({message: 'Artwork not found.'})
+      return err
+    }
+  }
+
+  onNewImage = (image) => {
+    const { section } = this.props
+    const newImages = clone(section.get('images')).concat(image)
+
+    section.set('images', newImages)
   }
 
   onUpload = (image, width, height) => {
-    let newImages = clone(this.props.images)
-    newImages = newImages.concat({
+    this.onNewImage({
       url: image,
       type: 'image',
       width: width,
       height: height,
       caption: ''
     })
-    this.props.section.set('images', newImages)
   }
 
-  onChangeImageSetTitle = (e) => {
-    this.props.section.set('title', e.target.value)
-  }
-
-  toggleImagesetLayout = (layout) => {
-    this.props.section.set('layout', layout)
-  }
-
-  inputsAreDisabled (section) {
+  inputsAreDisabled = () => {
+    const { section } = this.props
     return section.get('layout') === 'fillwidth' && section.get('images').length > 0
   }
 
-  fillwidthAlert () {
-    alert('Fullscreen layouts accept one asset, please remove extra images.')
+  fillwidthAlert = () => {
+    const { logErrorAction } = this.props
+    const message = 'Fullscreen layouts accept one asset, please remove extra images or use another layout.'
+
+    logErrorAction({ message })
```

Comment:
Tangent, but it would be nice to work with @owendodd at some point to design a less obstructive error notification UI. The flash might be a little jarring while editing at high speeds. Maybe with Portals 😛

Reply:
Seems like something we could definitely think about. Logged here: https://github.com/artsy/publishing/issues/309